test/minter-suite-minters/DA/MinterDAExp/MinterDAExpV3.test.ts

```diff
@@ -0,0 +1,307 @@
+        this.genArt721Core.address,
+        this.minterFilter.address,
+      ]);
+
+      await safeAddProject(
+        this.genArt721Core,
+        this.accounts.deployer,
+        this.accounts.artist.address
+      );
+
+      await this.genArt721Core
+        .connect(this.accounts.deployer)
+        .toggleProjectIsActive(this.projectZero);
+
+      await this.genArt721Core
+        .connect(this.accounts.artist)
+        .updateProjectMaxInvocations(this.projectZero, 15);
+
+      await this.genArt721Core
+        .connect(this.accounts.artist)
+        .toggleProjectIsPaused(this.projectZero);
+
+      await this.minterFilter
+        .connect(this.accounts.deployer)
+        .addApprovedMinter(this.minter.address);
+      await this.minterFilter
+        .connect(this.accounts.deployer)
+        .setMinterForProject(this.projectZero, this.minter.address);
+
+      if (!this.startTime) {
+        const blockNumber = await ethers.provider.getBlockNumber();
+        const block = await ethers.provider.getBlock(blockNumber);
+        this.startTime = block.timestamp;
+      }
+      this.startTime = this.startTime + ONE_DAY;
+
+      await ethers.provider.send("evm_mine", [this.startTime - ONE_MINUTE]);
+      await this.minter
+        .connect(this.accounts.deployer)
+        .resetAuctionDetails(this.projectZero);
+      await this.minter
+        .connect(this.accounts.artist)
+        .setAuctionDetails(
+          this.projectZero,
+          this.startTime + this.auctionStartTimeOffset,
+          this.defaultHalfLife,
+          this.startingPrice,
+          this.basePrice
+        );
+      await ethers.provider.send("evm_mine", [this.startTime]);
+    });
+
+    describe("common DAEXP tests", async function () {
+      await MinterDAExp_Common();
+    });
+
+    describe("common DA V1V2 tests", async function () {
+      await MinterDAV1V2_Common();
+    });
+
+    describe("setAuctionDetails", async function () {
+      it("does not unsafely cast auction time start > type(uint64).max", async function () {
+        await this.minter
+          .connect(this.accounts.deployer)
+          .resetAuctionDetails(this.projectZero);
+        const overflowStartTime = ethers.BigNumber.from("2").pow(
+          ethers.BigNumber.from("64")
+        );
+        await expectRevert(
+          this.minter
+            .connect(this.accounts.artist)
+            .setAuctionDetails(
+              this.projectZero,
+              overflowStartTime,
+              this.defaultHalfLife,
+              this.startingPrice,
+              this.basePrice
+            ),
+          "SafeCast: value doesn't fit in 64 bits"
+        );
+      });
+    });
+
+    describe("setProjectMaxInvocations", async function () {
+      it("allows artist to call setProjectMaxInvocations", async function () {
+        await this.minter
+          .connect(this.accounts.artist)
+          .setProjectMaxInvocations(this.projectZero);
+      });
+
```

Comment:
these two tests below are testing the logic around setting/reseting maxHasBeenInvoked accordingly when either setMaxProjectInvocations or manuallyLimitProjectMaxInvocations is called

Reply:
LGTM!